lib/rucio/core/rule.py

```diff
@@ -1911,6 +1921,21 @@ def generate_rule_notifications(rule, session=None):
                     except DataIdentifierNotFound:
                         pass
 
+    elif rule.state == RuleState.REPLICATING and rule.notification == RuleNotification.PROGRESS and replicating_locks_before:
+        # For RuleNotification PROGRESS rules, also notifiy when REPLICATING thresholds are passed
+        if __progress_class(replicating_locks_before, total_locks) != __progress_class(rule.locks_replicating_cnt, total_locks):
+            try:
+                did = rucio.core.did.get_did(scope=rule.scope, name=rule.name, session=session)
+                if not did['open']:
```

Comment:
I thought similar to CLOSE, PROGRESS also only applies for closed datasets. Otherwise there might be this weird case where a rule flip/flops between under/over a threshold (e.g. 50%)